app/models/classified_listing.rb

```diff
@@ -39,41 +24,45 @@ class ClassifiedListing < ApplicationRecord
   validates :user_id, presence: true
   validates :organization_id, presence: true, unless: :user_id?
 
-  validates :title, presence: true,
-                    length: { maximum: 128 }
-  validates :body_markdown, presence: true,
-                            length: { maximum: 400 }
+  validates :title, presence: true, length: { maximum: 128 }
+  validates :body_markdown, presence: true, length: { maximum: 400 }
   validates :location, length: { maximum: 32 }
   validate :restrict_markdown_input
   validate :validate_tags
   validate :validate_category
 
   scope :published, -> { where(published: true) }
 
-  def self.cost_by_category(category)
-    categories_available.dig(category, :cost) || 0
-  end
-
-  def author
-    organization || user
-  end
-
+  # TODO: refactor this class method block
   def self.select_options_for_categories
-    categories_available.keys.map do |key|
-      category = categories_available[key]
-      cost = category[:cost]
-      ["#{category[:name]} (#{cost} #{'Credit'.pluralize(cost)})", key]
+    ClassifiedListingCategory.select(:id, :name, :cost).map do |cl|
+      ["#{cl.name} (#{cl.cost} #{'Credit'.pluralize(cl.cost)})", cl.id]
     end
   end
 
   def self.categories_for_display
-    categories_available.keys.map do |key|
-      { slug: key, name: categories_available[key][:name] }
+    ClassifiedListingCategory.pluck(:slug, :name).map do |slug, name|
+      { slug: slug, name: name }
     end
   end
 
   def self.categories_available
-    CATEGORIES_AVAILABLE
+    ClassifiedListingCategory.all.each_with_object({}) do |cat, h|
+      h[cat.slug] = cat.attributes.slice("cost", "name", "rules")
+    end.deep_symbolize_keys
+  end
+
+  def category
+    classified_listing_category&.slug
+  end
+
+  def cost
```

Comment:
🔥

Reply:
Should we cache this to avoid the lookup multiple times?
```ruby
@cost ||= classified_listing_category&.cost || ClassifiedListingCategory.select(:cost).find_by(slug: category)&.cost
```

Reply:
Where we use it (mostly in `ClassifiedListingsToolkit`) it's generally assigned to a variable that will be checked. I can add it though. The only potential problem is that if we change the price anyone who started editing a listing just before will spend the old amount, but that's unlikely I guess.